pkg/scheduler/cache/cache.go

```diff
@@ -549,6 +549,10 @@ func (sc *SchedulerCache) Snapshot() *kbapi.ClusterInfo {
 	}
 
 	for _, value := range sc.Nodes {
+		if !value.Ready() {
```

Comment:
Do we always track unready nodes? I am thinking if scheduler should delete node over timeout?

Reply:
Good point ! We should delete node if there's no pod on it when timeout :) I'm going to open a separate PR for that.